### PR TITLE
Sync fan device type with specs

### PR DESCRIPTION
**Pull request by jadhavrohit924**
#### Problem
What is being fixed?  Examples:
* Fan device type was not in xml

#### Change overview
Added fan device type to sync with specs

#### Testing
How was this tested? (at least one bullet point required)
* Verified fan device type syncs with specs

### CODE REVIEW

File: src/app/zap-templates/zcl/data-model/chip/matter-devices.xml [hunk 1]
```diff
@@ -1854,6 +1854,45 @@ limitations under the License.
             <include cluster="Relative Humidity Measurement" client="false" server="false" clientLocked="false" serverLocked="false"></include>
         </clusters>
     </deviceType>
+    <deviceType>
+        <name>MA-fan</name>
+        <domain>HA</domain>
+        <typeName>Matter Fan</typeName>
+        <profileId editable="false">0x0103</profileId>
+        <deviceId editable="false">0x003B</deviceId>
```
Comment (dhrishi) [inline src/app/zap-templates/zcl/data-model/chip/matter-devices.xml:1862]: I see the value as 0x002B in the spec. Can you please confirm?

File: src/app/zap-templates/zcl/data-model/chip/matter-devices.xml [hunk 2]
```diff
@@ -1854,6 +1854,45 @@ limitations under the License.
...
+                <requireCommand>IdentifyQuery</requireCommand>
+                <requireCommand>TriggerEffect</requireCommand>
+            </include>
+            <include cluster="Descriptor" client="false" server="true" clientLocked="true" serverLocked="true">
+                <requireAttribute>DEVICE_TYPE_LIST</requireAttribute>
+                <requireAttribute>SERVER_LIST</requireAttribute>
+                <requireAttribute>CLIENT_LIST</requireAttribute>
+                <requireAttribute>PARTS_LIST</requireAttribute>
+            </include>
+            <include cluster="Binding" client="false" server="false" clientLocked="true" serverLocked="true">
```
Comment (dhrishi) [inline src/app/zap-templates/zcl/data-model/chip/matter-devices.xml:1877]: Why is the binding added here? Both server and client are set to false and locked so why are they required here?

File: src/app/zap-templates/zcl/data-model/chip/matter-devices.xml (hunk 1 above)
Comment (jadhavrohit924) [reply src/app/zap-templates/zcl/data-model/chip/matter-devices.xml]: Oops!! Updated to 0x002B. Thank you

File: src/app/zap-templates/zcl/data-model/chip/matter-devices.xml (hunk 2 above)
Comment (jadhavrohit924) [reply src/app/zap-templates/zcl/data-model/chip/matter-devices.xml]: Removed

File: src/app/zap-templates/zcl/data-model/chip/matter-devices.xml [hunk 3]
```diff
@@ -1854,6 +1854,42 @@ limitations under the License.
             <include cluster="Relative Humidity Measurement" client="false" server="false" clientLocked="false" serverLocked="false"></include>
         </clusters>
     </deviceType>
+    <deviceType>
+        <name>MA-fan</name>
+        <domain>HA</domain>
```
Comment (Damian-Nordic) [inline src/app/zap-templates/zcl/data-model/chip/matter-devices.xml:1859]: CHIP?

File: src/app/zap-templates/zcl/data-model/chip/matter-devices.xml [hunk 4]
```diff
@@ -1854,6 +1854,42 @@ limitations under the License.
...
+        <typeName>Matter Fan</typeName>
+        <profileId editable="false">0x0103</profileId>
+        <deviceId editable="false">0x002B</deviceId>
+        <clusters lockOthers="true">
+            <include cluster="Identify" client="false" server="true" clientLocked="true" serverLocked="true">
+                <requireAttribute>IDENTIFY_TIME</requireAttribute>
+                <requireAttribute>IDENTIFY_TYPE</requireAttribute>
+                <requireCommand>Identify</requireCommand>
+                <requireCommand>IdentifyQuery</requireCommand>
```
Comment (Damian-Nordic) [inline src/app/zap-templates/zcl/data-model/chip/matter-devices.xml:1868]: Is this correct? `Query` feature has conformance `!Matter` for Fan

File: src/app/zap-templates/zcl/data-model/chip/matter-devices.xml (hunk 3 above)
Comment (jadhavrohit924) [reply src/app/zap-templates/zcl/data-model/chip/matter-devices.xml]: Updated

File: src/app/zap-templates/zcl/data-model/chip/matter-devices.xml (hunk 4 above)
Comment (jadhavrohit924) [reply src/app/zap-templates/zcl/data-model/chip/matter-devices.xml]: Right, updated

**Review by Damian-Nordic — APPROVED** [review]

**Review by woody-apple — APPROVED** [review]

**Review by dhrishi — APPROVED** [review]

**Pull request merged**